src/main/java/edu/kis/powp/jobs2d/command/gui/ComplexCommandEditorWindow.java

```diff
@@ -0,0 +1,249 @@
+package edu.kis.powp.jobs2d.command.gui;
+
+import edu.kis.powp.appbase.gui.WindowComponent;
+import edu.kis.powp.jobs2d.command.*;
+import edu.kis.powp.jobs2d.command.complexCommandEditor.ComplexCommandEditor;
+import edu.kis.powp.jobs2d.command.complexCommandEditor.IComplexCommandEditor;
+import edu.kis.powp.jobs2d.features.CommandsFeature;
+
+import java.util.ArrayList;
+import javax.swing.*;
+import javax.swing.event.ListSelectionEvent;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+
+public class ComplexCommandEditorWindow extends JFrame implements WindowComponent {
+
+	private final JLabel commandNameValue;
+	private final JTextField paramXInput;
+	private final JTextField paramYInput;
+	private final JList<DriverCommand> commandList;
+	private final DefaultListModel<DriverCommand> listModel;
+	private final JLabel numberOfCommandsValue;
+	private final JLabel commandLengthValue;
+
+	private IComplexCommandEditor complexCommandEditor;
+	private ICompoundCommand currentCommand;
+
+	public ComplexCommandEditorWindow() {
+
```

Comment:
This method is too large. It should be decomposed to several methods to make it easier to understand and handle.